feat: update profile settings

amadeu2 opened the pull request:
## Summary
Please summarize your changes, describing __what__ they are and __why__ they were made.

### Changelog
```
Please write an accurate and concise changelog for your changes.
```

## Relevant Issues
Closes #3213 
Closes #3214
Closes #3215

## Type of Change
Please select any type below that is applicable to your changes, and delete those that are not.
- [ ] __Breaking__ - any change that would cause existing functionality to not work as expected
- [ ] __Chore__ - refactoring, build scripts or anything else that isn't user-facing
- [ ] __Docs__ - changes to the documentation
- [ ] __Fix__ - a change which fixes an issue
- [ ] __New__ - a change which implements a new feature
- [ ] __Update__ - a change which updates existing functionality

## Testing
### Platforms
Please select any platforms where your changes have been tested.
- __Desktop__
	- [ ] MacOS
	- [x] Linux
	- [ ] Windows
- __Mobile__
	- [x] iOS
	- [x] Android

### Instructions
Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

## Checklist
Please tick all of the following boxes that are relevant to your changes, and delete those that are not.
- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

paul-boegelsack commented on packages/shared/components/MainMenu.svelte:
```diff
@@ -1,78 +1,78 @@
-    function handleBackClick() {
-        if ($settingsRoute === SettingsRoute.Init) {
-            drawer?.close()
+    async function handleBackClick() {
```
Hasn't to be async?

amadeu2 replied:
fixed thanks!

paul-boegelsack commented on packages/shared/components/MainMenu.svelte:
```diff
@@ -1,78 +1,78 @@
-    import { settingsRoute, settingsRouter, SettingsRoute } from '@core/router'
-    import { Locale } from '@core/i18n'
+    import { 
+        profileRoute, 
+        profileRouter, 
+        ProfileRoute, 
+        settingsRoute, 
+        settingsRouter, 
+        SettingsRoute 
```
Get linting errors because of trailing spaces

amadeu2 replied:
🙏 done!

paul-boegelsack reviewed (commented):
I merged the bottom navigation fix, so maybe the TransactionTabs conflict can be overwritten with the current state on develop? 

Apart from that, just two minor things

paul-boegelsack commented on packages/shared/components/ProfileActions.svelte:
```diff
@@ -0,0 +1,234 @@
+        await logout()
+    }
+
+    function handleNetworkStatusClick() {
```
Missing void return type

paul-boegelsack commented on packages/shared/components/ProfileActions.svelte:
```diff
@@ -0,0 +1,234 @@
+        }
+    }
+
+    function handleBackupClick() {
```
Return type

paul-boegelsack commented on packages/shared/components/ProfileActions.svelte:
```diff
@@ -0,0 +1,234 @@
+        })
+    }
+
+    function handleVersionUpdateClick() {
```
Return type

paul-boegelsack commented on packages/shared/components/Drawer.svelte:
```diff
@@ -5,7 +5,7 @@
 	
 	@prop {boolean} [opened] - Opens drawer on load.
-	@prop {boolean} [fromRight] - Slide from right side.
+	@prop {boolean} [fromLeft] - Slide from right side.
```
```suggestion
	@prop {boolean} [fromLeft] - Slide from left side.
```

paul-boegelsack commented on packages/shared/components/MainMenu.svelte:
```diff
@@ -1,78 +1,75 @@
         drawer.open()
     }
+
+    function handleSettingsClick() {
```
return types

amadeu2 replied:
Added, but I think we could avoid to type `void` since the type is also introspected by Typescript without adding it.

paul-boegelsack reviewed (approved):
Just some refactorings.

paul-boegelsack reviewed (changes requested):
Realised the close button of the settings menu isn't working on desktop. Can you check this?
![image](https://user-images.githubusercontent.com/6748220/173825758-204f80bd-c237-4ddd-a2f3-5e8f478ff3f7.png)

amadeu2 commented on packages/shared/routes/dashboard/settings/Settings.svelte:
```diff
@@ -24,6 +25,7 @@
         SettingsRoute.Init && 'md:pt-20'} {$mobile && 'overflow-y-auto'} "
+    in:fly={{ duration: $mobile ? 200 : 0, x: 200 }}
 >
     {#if !$mobile}
```
> Realised the close button of the settings menu isn't working on desktop. Can you check that? ![image](https://user-images.githubusercontent.com/6748220/173825758-204f80bd-c237-4ddd-a2f3-5e8f478ff3f7.png)
> 
> PS: Don't know if that is related to the changes

Here we dont touched that code, so it's not the problem of this PR I think.

amadeu2 commented:
> I merged the bottom navigation fix, so maybe the TransactionTabs conflict can be overwritten with the current state on develop?
> 
> Apart from that, just two minor things
> 
> Edit: We have to add the related issues.

The issues with locales should be solved once 1.5 will be merged, we should finish this PR asap, I just added a new task for this.

Also I added the latest TransactionTabs updates from the latest alpha release.
